src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,41 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationFailed;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int CORRECT_LOGIN_PASSWORD_SIZE = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null || user.getPassword() == null || user.getAge() == null) {
+            throw new RegistrationFailed("login, password, age can't be " + null);
+        }
+
+        if (user.getLogin().length() < CORRECT_LOGIN_PASSWORD_SIZE) {
+            throw new RegistrationFailed("login leght must be bigger than --> "
+                    + CORRECT_LOGIN_PASSWORD_SIZE
+                    + " symbols");
```

Comment:
Typo: 'leght' should be 'length'. Also, the error message should be more descriptive. Instead of 'login leght must be bigger than -->', consider 'Login length must be at least 6 characters'.